packages/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy/lib/LinkQueueLogger.ts

```diff
@@ -0,0 +1,208 @@
+import type { ILinkQueue, ILink } from '@comunica/bus-rdf-resolve-hypermedia-links-queue';
+import { LinkQueueWrapper } from '@comunica/bus-rdf-resolve-hypermedia-links-queue';
+import type { Logger } from '@comunica/types';
+import { PRODUCED_BY_ACTOR } from '@comunica/types-link-traversal';
+import type { Algebra } from 'sparqlalgebrajs';
+
+/**
+ * The type of event
+ */
+export enum EventType {
+  Push,
+  Pop,
+}
+
+/**
+ * A link queue event
+ */
+interface ILinkQueueEvent {
+  type: EventType;
+  link: IURLStatistic;
+  query: Algebra.Operation;
+  queueStatistics: IQueueStatistics;
+}
+/**
+ * Statistic of the link queue
+ */
+interface IQueueStatistics {
+  size: number;
+  reachabilityRatio: IReachabilityRatio;
+}
+/**
+ * Ratio of the reachability criteria of the link in the queue
+ */
+interface IReachabilityRatio {
+  pushEvent: Record<string, number>;
+  popEvent: Record<string, number>;
+}
+/**
+ * Optional parameters necessitating special processing
+ */
+interface IOptionalLinkQueueParameters {
+}
```

Comment:
Is there a reason these are at the top of the file? I think most actors place their interfaces below the actor class